### PR TITLE
Added resources for question model

### PR DESCRIPTION
**Pull request by duckmoll**
#43 
Added API Resource to Question Models.

### CODE REVIEW

File: new_dawn_server/questions/api/resources.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        allowed_methods = ["get"]
+        queryset = Question.objects.all()
+        resource_name = "question"
+
+
+class AnswerQuestionResource(ModelResource):
+    question = fields.ForeignKey(QuestionResource, "question", related_name="answer_question", full=True)
+    user = fields.ToOneField(UserResource, "user", related_name="answer_question", full=True)
+
+    class Meta:
```
Comment (tangziyi001) [inline new_dawn_server/questions/api/resources.py]: Should allow both GET (get answer) and POST (answer question)
Also try adding some tests for GET and POST.
The POST test should be similar to my User register test.
For GET test, it should just involve getting the data and do some validation.
Try figure out it with tastypie test doc.

**Comment by duckmoll** [comment]
The overwrite of obj_create() function was awesome!

File: new_dawn_server/questions/api/resources.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "question": True,
+    "sample_answer": True,
+}
+
+
+class QuestionResource(ModelResource):
+    class Meta:
+        always_return_data = True
+        authorization = Authorization()
+        allowed_methods = ["get", "post"]
```
Comment (tangziyi001) [inline new_dawn_server/questions/api/resources.py]: Not sure if we want user to be able to create questions. But we can keep this for now.

File: new_dawn_server/questions/api/tests/test_questions_api.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        self.assertEqual(AnswerQuestion.objects.count(), 0)
+        all_arguments = {
+            **self.answer_question_argument,
+            "user": "/api/v1/user/1/",
+            "question": "/api/v1/question/1/"
+        }
+        self.api_client.post(
+            "/api/v1/question/", format="json", data=self.question_argument)
+
+        self.api_client.post(
```
Comment (tangziyi001) [inline new_dawn_server/questions/api/tests/test_questions_api.py]: Why you need to do this given that a user has already registered in def setUP (which creates user, account and profile altogether)?

File: new_dawn_server/questions/api/tests/test_questions_api.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+        all_arguments = {
+            **self.answer_question_argument,
+            "user": "/api/v1/user/1/",
+            "question": "/api/v1/question/1/"
+        }
+        self.api_client.post(
+            "/api/v1/question/", format="json", data=self.question_argument)
+
+        self.api_client.post(
```
Comment (tangziyi001) [inline new_dawn_server/questions/api/tests/test_questions_api.py]: same as comment above

File: new_dawn_server/questions/api/tests/test_questions_api.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+        res = self.api_client.post(
+            "/api/v1/answer_question/", format="json", data=all_arguments)
+        res_data = json.loads(res.content)
+
+        for k, v in all_arguments.items():
+            self.assertTrue(k in res_data)
+            # Already tested user and question
+            if k == "user" or k == "question":
+                continue
```
Comment (tangziyi001) [inline new_dawn_server/questions/api/tests/test_questions_api.py]: can you confirm that user and question info is also in the res_data?

Comment (tangziyi001) [reply new_dawn_server/questions/api/tests/test_questions_api.py]: i mean the actual fields (username, etc.)

File: new_dawn_server/questions/api/tests/test_questions_api.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            "/api/v1/answer_question/", format="json", data=all_arguments)
+        res_data = json.loads(res.content)
+
+        for k, v in all_arguments.items():
+            self.assertTrue(k in res_data)
+            # Already tested user and question
+            if k == "user":
+                no_password_args = self.register_argument
+                no_password_args.pop("password", None)
+                self.assertTrue(no_password_args.items() < res_data["user"].items())
```
Comment (tangziyi001) [inline new_dawn_server/questions/api/tests/test_questions_api.py]: Why not just iterate and check the item values are identical?
This "<" check doesn't tell anything about the response data. People looking at this part will have no clue about what this is doing

**Pull request merged**